python/packages/autogen-core/src/autogen_core/tools/_base.py

```diff
@@ -147,8 +148,16 @@ def return_value_as_string(self, value: Any) -> str:
     async def run(self, args: ArgsT, cancellation_token: CancellationToken) -> ReturnT: ...
 
     async def run_json(self, args: Mapping[str, Any], cancellation_token: CancellationToken) -> Any:
-        # Execute the tool's run method
-        return_value = await self.run(self._args_type.model_validate(args), cancellation_token)
+        with get_tracer("base_tool").start_as_current_span(
```

Comment:
Doesn't seem the `get_current_span` is used here? Should we be using the same tracer created in the `TraceHelper` class?

Reply:
From my testing that's what it does, check the picture in the description. Essentially the call to `start_as_current_span` will check for an existing `span` in the `Context` and use that if no span or context is provided directly.

Reply:
Tested. Looks good.